app/routes.py

```diff
@@ -1,2 +1,286 @@
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'},200)
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        response_value = {"goal":{
+            "id": goal.goal_id,
+            "title": goal.title,
+        }}
+
+        return make_response(response_value, 200)
+
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            if not task.completed_at:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                })
+            else:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.completed_at
+                })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if request_body.get("title") and request_body.get("description"): 
+            if request_body.get("completed_at")==None:
+                new_task = Task(title=request_body["title"],
+                                description=request_body["description"])
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": False
+                }}
+            elif request_body.get("completed_at"):
+                new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=datetime.now())
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": True
+                    }}
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
```

Comment:
Using that ternary mentioned earlier, we can get rid of a lot of repetitive lines of code and refactor to:

```suggestion
        if request_body.get("title") and request_body.get("description") and "completed_at" in request_body : 
                new_task = Task(title=request_body["title"],
                                description=request_body["description"])
                db.session.add(new_task)
                db.session.commit()
                response_value = {"task":{
                        "id": new_task.task_id,
                        "title": new_task.title,
                        "description": new_task.description,
                        "is_complete": True if task.completed_at else False
                }}
        else:
            return make_response({"details":"Invalid data"}, 400)

```

Reply:
Preferably, we could use a guard clause to check for the failed case (invalid data) early  to get rid of the dangly one-line `else` clause.